### PR TITLE
feat(core, webserver): add /v1beta/server_setting api

### PR DESCRIPTION
**Pull request by wsxiaoys**

### CODE REVIEW

File: ee/tabby-webserver/src/handler.rs [hunk 1]
```diff
@@ -72,3 +77,19 @@ async fn distributed_tabby_layer(
...
+    let setting = match locator.setting().read_security_setting().await {
+        Ok(x) => x,
+        Err(err) => {
+            warn!("Failed to read security setting {}", err);
+            return Err(StatusCode::INTERNAL_SERVER_ERROR);
+        }
+    };
+
+    Ok(Json(ServerSetting {
+        disable_client_side_telemetry: setting.disable_client_side_telemetry,
```
Comment (boxbeam) [inline ee/tabby-webserver/src/handler.rs:93]: If this is to convert the DAO, don't we have a `From` impl for it?

**Review by boxbeam — APPROVED** [review]
Otherwise LGTM

File: ee/tabby-webserver/src/handler.rs (hunk 1 above)
Comment (wsxiaoys) [reply ee/tabby-webserver/src/handler.rs:93]: mostly not wanna create a practice that whatever in `ServerSetting` should be stored in `ServerSetting` table.

They should be somewhat irrelevant concept, but unfortunately reused the name :/

Comment (boxbeam) [reply ee/tabby-webserver/src/handler.rs:93]: Just concerned about the number of places we might need to duplicate this logic if we won't be using into() - do you mean you want to decouple the DAO and external representation more?

**Pull request merged**